tests/integration/apigateway/test_apigateway_api.py

```diff
@@ -1,16 +1,265 @@
 
+import pytest
+from botocore.exceptions import ClientError
+
+from localstack.services.apigateway.helpers import TAG_KEY_CUSTOM_ID
+from localstack.testing.aws.util import is_aws_cloud
 from localstack.utils.files import load_file
+from localstack.utils.strings import short_uid
+
+LOG = logging.getLogger(__name__)
 
 # parent directory of this file
 PARENT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 OPENAPI_SPEC_PULUMI_JSON = os.path.join(PARENT_DIR, "files", "openapi.spec.pulumi.json")
 
 
+@pytest.fixture(autouse=True)
+def apigw_snapshot_transformer(snapshot):
+    snapshot.add_transformer(snapshot.transform.apigateway_api())
+
+
+@pytest.fixture(scope="class", autouse=True)
+def apigw_cleanup_before_run(apigateway_client):
+    # TODO: remove this once all tests are properly cleaning up and using fixtures
+    rest_apis = apigateway_client.get_rest_apis()
+    for rest_api in rest_apis["items"]:
+        delete_rest_api_retry(apigateway_client, rest_api["id"])
+
+
+def delete_rest_api_retry(client, rest_api_id: str):
+    try:
+        if is_aws_cloud():
+            # This is ugly but API GW returns 429 very quickly, and we want to be sure to clean up properly
+            cleaned = False
+            while not cleaned:
+                try:
+                    client.delete_rest_api(restApiId=rest_api_id)
+                    cleaned = True
+                except ClientError as e:
+                    error_message = str(e)
+                    if "TooManyRequestsException" in error_message:
+                        time.sleep(10)
+                    elif "NotFoundException" in error_message:
+                        break
+                    else:
+                        raise
+        else:
+            client.delete_rest_api(restApiId=rest_api_id)
+
+    except Exception as e:
+        LOG.debug("Error cleaning up rest API: %s, %s", rest_api_id, e)
+
+
+@pytest.fixture
+def apigw_create_rest_api(apigateway_client):
+    rest_apis = []
+
+    def _factory(*args, **kwargs):
+        if "name" not in kwargs:
+            kwargs["name"] = f"test-api-{short_uid()}"
+        response = apigateway_client.create_rest_api(*args, **kwargs)
+        rest_apis.append(response["id"])
+        return response
+
+    yield _factory
+
+    # TODO: might clean up even more resources as we learn? integrations and such?
+    for rest_api_id in rest_apis:
+        delete_rest_api_retry(apigateway_client, rest_api_id)
+
+
 def test_import_rest_api(import_apigw, snapshot):
     snapshot.add_transformer(snapshot.transform.apigateway_api())
 
     spec_file = load_file(OPENAPI_SPEC_PULUMI_JSON)
     response, root_id = import_apigw(body=spec_file, failOnWarnings=True)
 
     snapshot.match("import_rest_api", response)
+
+
+class TestApiGatewayApi:
+    @pytest.mark.aws_validated
+    def test_list_and_delete_apis(self, apigateway_client, apigw_create_rest_api, snapshot):
+        api_name1 = "test-list-and-delete-apis-1"
```

Comment:
nit: I'd suggest to better randomize the resource names here. These names would probably be sufficiently distinct, but seems like a good practice to randomize resource names, to avoid potential name clashes going forward.. (e.g., if we're performing lookups based on names) 👍

Reply:
Agreed! Even though API names don't need to be distinct and we can have several with the same names, so no clash could happen while trying to delete, and we can't look them up by name. I had this set up at first because I hadn't put the name in the snapshot transformer, as I didn't see the need. But in the base transformer already set up, it was there. I'll add randomness there so we're sure, but it shouldn't have created any issue!